salt/kube-apiserver/init.sls

```diff
@@ -112,9 +112,3 @@ kube-apiserver:
       - file:     kube-apiserver
       - sls:      ca-cert
       - {{ pillar['ssl']['kube_apiserver_crt'] }}
-  # wait for the API server (see k8s issue #47739) so it can be used
-  http.wait_for_successful_query:
```

Comment:
Oh, why did we remove this? It's better to keep this with kube-apiserver, so other things which need it (e.g. the CNI flannel change) can reuse it.

Reply:
... and you could remove [this loop](https://github.com/kubic-project/salt/pull/193/files#diff-3d900be66a547ca3549316625a8e2dd5R24).

Reply:
@kiall @inercia The http.wait_for_successful_query state doesn't work correctly ever since we added haproxy in front of the kube-api servers. We were running into an issues where http.wait_for_successful_query would pass (i.e. the kube-api server would report as ready when it was hit directly on the unauthenticated localhost port on 8080) but going through the local haproxy would error out (presumably because no backends were available yet -- not having passed the healthy interval check). 

To fix this, we updated the http.wait_for_successful_query call with that curl loop that actually makes an authenticated call through haproxy, which ensures that kube-api is actually up before we make issue any kubectl commands. 

It might be a good idea to add the curl check to this state though -- so that we don't break if another api-dependent state is introduced between this and the place where the check exists today.

Reply:
Why don't we run `wait_for_successful_query` on the _haproxy_ port? I'm sure haproxy will return an error if the backend (the API server) is not ready yet. I guess that could be reliable enough...

Reply:
`wait_for_successful_query` doesn't accept params for the client certs, so, as long as the authenticated port is used, it will fail.

Since there is a delay (on a per-backend basis) for haproxy to healthcheck nodes, the only reliable check is to check the authenticated port.

Reply:
`wait_for_successful_query` seems to support certificates: [this function](https://github.com/saltstack/salt/blob/d1f80bab3993059060ca51263e9d048e3a618051/salt/modules/http.py#L41) will eventually call [salt.utils.http.query](https://github.com/saltstack/salt/blob/develop/salt/utils/http.py#L98), a function that can accept a [CA](https://github.com/saltstack/salt/blob/develop/salt/utils/http.py#L130) and a [certificate](https://github.com/saltstack/salt/blob/develop/salt/utils/http.py#L132) (looking at [these lines](https://github.com/saltstack/salt/blob/develop/salt/utils/http.py#L442), `cert` can be a tuple with `(cert, key)`).

I would give it a try, but if this is too much hassle now I would leave it for a subsequent PR...